### PR TITLE
PEP458: Add minimal RSTUF service configuration

### PR DESCRIPTION
**Pull request by kairoaraujo**
Supersedes: #13943

This PR is part 1 in a series of PRs to integrate Repository Service for TUF (RSTUF) with Warehouse for PEP 458 adoption. 

**Background**
RSTUF maintains a PEP 458 TUF metadata repository as a black box. It provides REST API, which accepts requests upon addition or removal of release files, and a Worker service, which performs the TUF metadata update and signing tasks asynchronously upon API requests.

**Change details**
- Add `rstuf-api` service to docker-compose.yml
  - pull pinned image
  - configure redis

- Add `rstuf-worker` service to docker-compose.yml
  - pull pinned image
  - configure storage backend for TUF metadata files
  - configure Postgres DB for TUF metadata
  - configure Redis, used as task message queue for communication between API and Worker, and between Workers

- Include rstuf db initialization in `initdb` target in Makefile.

**Preview of next PR**
The next PR will include:
- Signing configuration, to sign online metadata changes
- Static root metadata, to bootstrap the RSTUF repository
- A Makefile target, to trigger bootstrap
- Instructions, to interact with the RSTUF REST API

### CODE REVIEW

**Comment by kairoaraujo** [comment]
@lukpueh

File: docker-compose.yml [hunk 1]
```diff
@@ -8,15 +8,16 @@ volumes:
...
+  rstuf-metadata:
 
 services:
   vault:
     # NOTE: pinned for consistency with whats available in our deployment
     image: vault:1.12.3
     restart: on-failure
     entrypoint: /bin/sh
-    command: /etc/vault/entry.sh
-    stop_signal: SIGINT
```
Comment (lukpueh) [inline docker-compose.yml:19]: The `command` rewrites and `stop_signal` removals  (here and below) seem unrelated. Can we remove them from this PR?

Comment (kairoaraujo) [reply docker-compose.yml]: I had an outdated fork. Thanks.

File: docker-compose.yml [hunk 2]
```diff
@@ -160,6 +161,37 @@ services:
...
+    environment:
+      - RSTUF_STORAGE_BACKEND=LocalStorage
+      - RSTUF_LOCAL_STORAGE_BACKEND_PATH=/var/opt/repository-service-tuf/storage
+      - RSTUF_BROKER_SERVER=redis://redis/1
+      - RSTUF_REDIS_SERVER=redis://redis
+      - RSTUF_REDIS_SERVER_DB_RESULT=1
+      - RSTUF_REDIS_SERVER_DB_REPO_SETTINGS=2
+      - RSTUF_SQL_SERVER=postgresql://postgres@db:5432/rstuf
+    healthcheck:
+      test: "exit 0"
```
Comment (ewdurbin) [inline docker-compose.yml:188]: If the nest is a null op, is it necessary?

File: docker-compose.yml [hunk 3]
```diff
@@ -160,6 +161,37 @@ services:
...
+    restart: always
+    tty: true
```
Comment (ewdurbin) [inline docker-compose.yml:190]: curious why this is set.

File: docker-compose.yml [hunk 4]
```diff
@@ -160,6 +161,37 @@ services:
       ARCHIVE_FILES_BACKEND: "warehouse.packaging.services.LocalArchiveFileStorage path=/var/opt/warehouse/packages-archive/ url=http://files:9001/packages-archive/{path}"
       SIMPLE_BACKEND: "warehouse.packaging.services.LocalSimpleStorage path=/var/opt/warehouse/simple/ url=http://files:9001/simple/{path}"
 
+  rstuf-api:
+    image: ghcr.io/repository-service-tuf/repository-service-tuf-api:v0.9.0b1
```
Comment (ewdurbin) [inline docker-compose.yml:165]: Why do the versions of the service and worker differ?

**Review by ewdurbin — APPROVED** [review]
overall i appreciate how clean this is when treated as a dependent service.

i believe we could remove the `vault` service at this time since nothing integrates with it and it only was added in support of implementing TUF inside warehouse.

File: docker-compose.yml (hunk 2 above)
Comment (kairoaraujo) [reply docker-compose.yml:188]: It is not necessary. I'm removing it.

File: docker-compose.yml (hunk 3 above)
Comment (kairoaraujo) [reply docker-compose.yml:190]: Same as above. It is not necessary. I'm removing it.
(It came from my personal development environment -- not Warehouse development)

File: docker-compose.yml (hunk 4 above)
Comment (kairoaraujo) [reply docker-compose.yml:165]: The RSTUF API and Worker have independent development and releases.

**Comment by kairoaraujo** [comment]
> i believe we could remove the `vault` service at this time since nothing integrates with it and it only was added in support of implementing TUF inside warehouse.

I added the `vault` removal as the first commit to keep a logical sequence in the PR commits.

**Comment by ewdurbin** [comment]
Thanks @kairoaraujo! Looks great.

**Comment by ewdurbin** [comment]
Unclear to me why CI is failing. Seems to be unrelated and consistent with other branches.

**Comment by kairoaraujo** [comment]
> Unclear to me why CI is failing. Seems to be unrelated and consistent with other branches.

I see this failure in two other opened PRs 🤔

**Comment by kairoaraujo** [comment]
> > Unclear to me why CI is failing. Seems to be unrelated and consistent with other branches.
> 
> I see this failure in two other opened PRs 🤔
> 
> https://github.com/pypi/warehouse/actions/runs/7655175103/job/20867003665#step:8:1

I did some investigation and I commented here:
https://github.com/pypi/warehouse/pull/15276#issuecomment-1911775508
